### ISSUE TITLE
After checking file, sometimes redownloads anon mistakenly [RC6]

### ISSUE THREAD

**Issue by colin1497** [Tribler/tribler#1684]
Sometimes, I assume on a failed check, if the file is marked download public, seed anon, tribler will try to download anon instead of public.

Lots of information provided in #1605, splitting this issue out for tracking purposes.

**Comment by whirm** [comment]
You are talking about a failed file hash check, right?

**Comment by colin1497** [comment]
This is my assumption.  I really can't say exactly what happened, but of the 60+ torrents, a few decided to redownload after the upgrade instead of skipping straight to seeding.  At that point, they downloaded anon instead of BT.

**Comment by whirm** [comment]
@colin1497 could you see if they started from scratch or from near 100%?

**Comment by colin1497** [comment]
They started from scratch.

**Comment by whirm** [comment]
@colin1497 still no log in %APPDATA%?

**Comment by whirm** [comment]
Also, to verify that it still was happening on RC6, did you restore a 6.4.3 .Tribler dir and then upgraded to RC6? (Trying to make a picture of the situation here)

**Comment by colin1497** [comment]
No, this is going from RC5 to RC6.  Also, see what I just posted in #1669.  This torrent was added immediately after restarting the client.
![tribler](https://cloud.githubusercontent.com/assets/11151259/10818206/b02ff4cc-7e0d-11e5-9017-9021f084ac4a.png)

**Comment by colin1497** [comment]
OK, there was an error log before I started this process, but Tribler had been running for a long time and so I deleted it before restart because I didn't want to muddle these specific issues.

I removed the torrent (not data)
I exited tribler
deleted existing error log
started tribler
added the magnet link
made the above screen cap and post
exited tribler
No error log
started tribler

If you'd like, I can leave it running and provide whatever error log it provides next time just for reference.

**Comment by colin1497** [comment]
Below are the contents of the log I deleted above, for reference:

ERROR   1445974938.86             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 824, in runUntilCurrent
    
  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 200, in <lambda>
    
  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 247, in network_create_engine_wrapper
    
  File "Tribler\Core\Libtorrent\LibtorrentMgr.pyo", line 239, in add_torrent
    
exceptions.RuntimeError: torrent already exists in session

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 824, in runUntilCurrent
    
  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 200, in <lambda>
    
  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 247, in network_create_engine_wrapper
    
  File "Tribler\Core\Libtorrent\LibtorrentMgr.pyo", line 239, in add_torrent
    
exceptions.RuntimeError: torrent already exists in session
ERROR   1445979261.11             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446029810.88             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446041826.82             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446064516.44             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446069112.33             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446074866.71             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446093380.66             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446097175.28             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446102994.88             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446103469.09             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446103619.86             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple
ERROR   1446110862.45             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run
    
  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner
    
  File "twisted\internet\base.pyo", line 1192, in run
    
  File "twisted\internet\base.pyo", line 1201, in mainLoop
    
--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 797, in runUntilCurrent
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1068, in on_data
    
  File "Tribler\community\tunnel\tunnel_community.pyo", line 1229, in crypto_in
    
exceptions.TypeError: %d format: a number is required, not tuple

**Comment by colin1497** [comment]
OK, after step 9 above (last restart) it checked the file and changed the status to completed instead of trying to redownload it.

**Comment by whirm** [comment]
Oh, ok, that's a duplicate of #1612 and a typo on a log call. I've fixed the log call and will fix the other one ASAP with the info you provided.

Thanks a lot for your help reproducing this!

**Comment by colin1497** [comment]
No problem.  Hopefully that takes care of it.

**Comment by colin1497** [comment]
Well crud, just had this happen again on 3 torrents (they were flagged as Anon: no after a restart, see my comment in #1669 ):

https://github.com/Tribler/tribler/issues/1669#issuecomment-160475456

Is there a newer test build available?

**Comment by whirm** [comment]
I missed your comment somehow, 
Here you can get the latest pre-release branch build: https://jenkins.tribler.org/job/Publish_tribler_next/lastStableBuild/

**Comment by whirm** [comment]
@colin1497 btw, if you still get the error, please create a new issue for each stacktrace you get, it helps us a lot in keeping track of things.

**Comment by colin1497** [comment]
It's OK on missing the comment, you had actually linked to try a newer build and I'd rummaged around the Jenkins site and figured things out.  I'll grab today's next build.
